Sources/Common/Core/DataArray/index.js

```diff
@@ -307,6 +348,61 @@ function vtkDataArray(publicAPI, model) {
 
     return sortedObj;
   };
+
+  publicAPI.deepCopy = (other) => {
+    publicAPI.shallowCopy(other);
+    publicAPI.setData(other.getData().slice());
+  };
+
+  // TODO: Check
+  publicAPI.interpolateTuple = (
+    idx,
+    source1,
+    source1Idx,
+    source2,
+    source2Idx,
+    t
+  ) => {
+    const numberOfComponents = model.numberOfComponents || 1;
+    if (
+      numberOfComponents !== source1.getNumberOfComponents() ||
+      numberOfComponents !== source2.getNumberOfComponents()
+    ) {
+      vtkErrorMacro('numberOfComponents must match');
+    }
+
+    const tuple1 = [];
+    const tuple2 = [];
+    const out = [];
+    out.length = numberOfComponents;
+
+    source1.getTuple(source1Idx, tuple1);
+    source2.getTuple(source2Idx, tuple2);
+
+    // Check most common component sizes first
+    // to avoid doing a for loop if possible
+    if (numberOfComponents === 1) {
+      out[0] = tuple1[0] + (tuple2[0] - tuple1[0]) * t;
+    } else if (numberOfComponents === 2) {
+      out[0] = tuple1[0] + (tuple2[0] - tuple1[0]) * t;
+      out[1] = tuple1[1] + (tuple2[1] - tuple1[1]) * t;
+    } else if (numberOfComponents === 3) {
+      out[0] = tuple1[0] + (tuple2[0] - tuple1[0]) * t;
+      out[1] = tuple1[1] + (tuple2[1] - tuple1[1]) * t;
+      out[2] = tuple1[2] + (tuple2[2] - tuple1[2]) * t;
+    } else if (numberOfComponents === 4) {
+      out[0] = tuple1[0] + (tuple2[0] - tuple1[0]) * t;
+      out[1] = tuple1[1] + (tuple2[1] - tuple1[1]) * t;
+      out[2] = tuple1[2] + (tuple2[2] - tuple1[2]) * t;
+      out[3] = tuple1[3] + (tuple2[3] - tuple1[3]) * t;
+    } else {
+      for (let i = 0; i < numberOfComponents; i++) {
+        out[i] = tuple1[i] + (tuple2[i] - tuple1[i]) * t;
+      }
```

Comment:
```
switch(numberOfComponents) {
  case 4:
    out[3] = tuple1[3] + (tuple2[3] - tuple1[3]) * t;
  case 3:
    out[2] = tuple1[2] + (tuple2[2] - tuple1[2]) * t;
  case 2:
    out[1] = tuple1[1] + (tuple2[1] - tuple1[1]) * t;
  case 1:
    out[0] = tuple1[0] + (tuple2[0] - tuple1[0]) * t;
    break;
  default:
    for (let i = 0; i < numberOfComponents; i++) {
        out[i] = tuple1[i] + (tuple2[i] - tuple1[i]) * t;
      }
};
```

Reply:
Nice! Very elegant :)

Could be changed for `getTuple` too btw.